### PR TITLE
HW2_v1

### PR DESCRIPTION
**Pull request by NikonovaEkaterina**
HW2 version 1

### CODE REVIEW

File: src/test/java/hw2/ex1/DefaultPageTest.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        assertTrue(driver.findElement(By.cssSelector("ul.sidebar-menu>li:nth-child(1)>a>span")).isDisplayed());
+        assertEquals(driver.findElement(By.cssSelector("ul.sidebar-menu>li:nth-child(1)>a>span")).getText(),"Home");
+        assertTrue(driver.findElement(By.cssSelector("ul.sidebar-menu>li:nth-child(2)>a>span")).isDisplayed());
+        assertEquals(driver.findElement(By.cssSelector("ul.sidebar-menu>li:nth-child(2)>a>span")).getText(),"Contact form");
+        assertTrue(driver.findElement(By.cssSelector("ul.sidebar-menu>li:nth-child(3)>a>span")).isDisplayed());
+        assertEquals(driver.findElement(By.cssSelector("ul.sidebar-menu>li:nth-child(3)>a>span")).getText(),"Service");
+        assertTrue(driver.findElement(By.cssSelector("ul.sidebar-menu>li:nth-child(4)>a>span")).isDisplayed());
+        assertEquals(driver.findElement(By.cssSelector("ul.sidebar-menu>li:nth-child(4)>a>span")).getText(),"Metals & Colors");
+        assertTrue(driver.findElement(By.cssSelector("ul.sidebar-menu>li:nth-child(5)>a>span")).isDisplayed());
+        assertEquals(driver.findElement(By.cssSelector("ul.sidebar-menu>li:nth-child(5)>a>span")).getText(),"Elements packs");
```
Comment (komyagin) [inline src/test/java/hw2/ex1/DefaultPageTest.java]: Why don't you move repeated parts to a separate method?

File: src/test/java/hw2/ex1/DefaultPageTest.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        assertEquals(driver.findElement(By.cssSelector("div.col-sm-3:nth-child(1)>div.benefit>span.benefit-txt")).getText(),"To include good practices\n" +
+                "and ideas from successful\n" +
+                "EPAM project");
+        assertEquals(driver.findElement(By.cssSelector("div.col-sm-3:nth-child(2)>div.benefit>span.benefit-txt")).getText(),"To be flexible and\n" +
+                "customizable");
+        assertEquals(driver.findElement(By.cssSelector("div.col-sm-3:nth-child(3)>div.benefit>span.benefit-txt")).getText(),"To be multiplatform");
+        assertEquals(driver.findElement(By.cssSelector("div.col-sm-3:nth-child(4)>div.benefit>span.benefit-txt")).getText(),"Already have good base\n" +
+                "(about 20 internal and\n" +
+                "some external projects),\n" +
+                "wish to get more…");
```
Comment (komyagin) [inline src/test/java/hw2/ex1/DefaultPageTest.java]: Why don't you move repeated parts to a separate method?

File: src/test/java/hw2/ex1/DefaultPageTest.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        assertEquals(headers.size(),4);
+        assertTrue(driver.findElement(By.cssSelector("ul.uui-navigation.nav.navbar-nav.m-l8>li:nth-child(1)>a")).isDisplayed());
+        assertEquals(driver.findElement(By.cssSelector("ul.uui-navigation.nav.navbar-nav.m-l8>li:nth-child(1)>a")).getText(),"HOME");
+        assertTrue(driver.findElement(By.cssSelector("ul.uui-navigation.nav.navbar-nav.m-l8>li:nth-child(2)>a")).isDisplayed());
+        assertEquals(driver.findElement(By.cssSelector("ul.uui-navigation.nav.navbar-nav.m-l8>li:nth-child(2)>a")).getText(),"CONTACT FORM");
+        assertTrue(driver.findElement(By.cssSelector("ul.uui-navigation.nav.navbar-nav.m-l8>li:nth-child(3)>a")).isDisplayed());
+        assertEquals(driver.findElement(By.cssSelector("ul.uui-navigation.nav.navbar-nav.m-l8>li:nth-child(3)>a")).getText(),"SERVICE");
+        assertTrue(driver.findElement(By.cssSelector("ul.uui-navigation.nav.navbar-nav.m-l8>li:nth-child(4)>a")).isDisplayed());
+        assertEquals(driver.findElement(By.cssSelector("ul.uui-navigation.nav.navbar-nav.m-l8>li:nth-child(4)>a")).getText(),"METALS & COLORS");
```
Comment (komyagin) [inline src/test/java/hw2/ex1/DefaultPageTest.java]: Why don't you move repeated parts to a separate method?

File: src/test/java/hw2/ex1/DefaultPageTest.java [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+        //2. Assert Browser title
+        assertEquals(driver.getTitle(), "Home Page");
+
+        //3. Perform login
+        WebDriverWait waitcicon = new WebDriverWait(driver, 10);
+        waitcicon.until(ExpectedConditions.presenceOfElementLocated(By.id("user-icon")));
+        driver.findElement(By.id("user-icon")).click();
+        driver.findElement(By.id("name")).sendKeys("Roman");
+        driver.findElement(By.id("password")).sendKeys("Jdi1234");
+        driver.findElement(By.id("login-button")).click();
```
Comment (komyagin) [inline src/test/java/hw2/ex1/DefaultPageTest.java]: Maybe we can create a specific method for login?

File: src/test/java/hw2/ex1/DefaultPageTest.java [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+
+public class DefaultPageTest {
+
+    private WebDriver driver;
+
+    @BeforeMethod(alwaysRun = true)
+    public void ChromeSetUp()
+    {
+        driver = new ChromeDriver();
+    }
```
Comment (komyagin) [inline src/test/java/hw2/ex1/DefaultPageTest.java]: Maybe we can move this to a separate class?

File: src/test/java/hw2/ex2/DifElementsPageTest.java [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+public class DifElementsPageTest {
+
+    private WebDriver driver;
+
+    @BeforeMethod(alwaysRun = true)
+    public void ChromeSetUp()
+    {
+        driver = new ChromeDriver();
+    }
```
Comment (komyagin) [inline src/test/java/hw2/ex2/DifElementsPageTest.java]: We should move this to a separate class

File: src/test/java/hw2/ex2/DifElementsPageTest.java [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+        //9.1 Assert that
+        //•	for each checkbox there is an individual log row and value is corresponded to the status of checkbox
+        assertTrue(driver.findElement(By.xpath("//li[contains(text(),'Water')]")).isDisplayed());
+        assertTrue(driver.findElement(By.xpath("//li[contains(text(),'Wind')]")).isDisplayed());
+
+        //•	for radio button there is a log row and value is corresponded to the status of radio button
+        assertTrue(driver.findElement(By.xpath("//li[contains(text(),'Selen')]")).isDisplayed());
+
+        //•	for dropdown there is a log row and value is corresponded to the selected value
+        assertTrue(driver.findElement(By.xpath("//li[contains(text(),'Yellow')]")).isDisplayed());
```
Comment (komyagin) [inline src/test/java/hw2/ex2/DifElementsPageTest.java]: Why don't you move repeated parts to a separate method?

File: src/test/java/hw2/ex2/DifElementsPageTest.java [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+        //2. Assert Browser title
+        assertEquals(driver.getTitle(), "Home Page");
+
+        //3. Perform login
+        WebDriverWait waitcicon = new WebDriverWait(driver, 10);
+        waitcicon.until(ExpectedConditions.presenceOfElementLocated(By.id("user-icon")));
+        driver.findElement(By.id("user-icon")).click();
+        driver.findElement(By.id("name")).sendKeys("Roman");
+        driver.findElement(By.id("password")).sendKeys("Jdi1234");
+        driver.findElement(By.id("login-button")).click();
```
Comment (komyagin) [inline src/test/java/hw2/ex2/DifElementsPageTest.java]: Maybe we can create a specific method for login?

File: src/test/java/hw2/ex2/DifElementsPageTest.java [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+        driver.manage().window().maximize();
+
+        //1. Open test site by URL
+        driver.get("https://jdi-testing.github.io/jdi-light/index.html");
+
+        //2. Assert Browser title
+        assertEquals(driver.getTitle(), "Home Page");
+
+        //3. Perform login
+        WebDriverWait waitcicon = new WebDriverWait(driver, 10);
```
Comment (komyagin) [inline src/test/java/hw2/ex2/DifElementsPageTest.java]: Name entities of the class according to Java Convention using camelCase

File: src/test/java/hw2/ex2/DifElementsPageTest.java [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+        driver = new ChromeDriver();
+    }
+
+    @Test (description = "Different Elements Page Test")
+    public void MyTest_02()  {
+
+        driver.manage().window().maximize();
+
+        //1. Open test site by URL
+        driver.get("https://jdi-testing.github.io/jdi-light/index.html");
```
Comment (komyagin) [inline src/test/java/hw2/ex2/DifElementsPageTest.java]: We should move this to the string constant

File: src/test/java/hw2/ex1/DefaultPageTest.java [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+
+    @Test (description = "Default Page Test")
+    public void MyTes_01() {
+
+        driver.manage().window().maximize();
+
+        //1. Open test site by URL
+        driver.get("https://jdi-testing.github.io/jdi-light/index.html");
```
Comment (komyagin) [inline src/test/java/hw2/ex1/DefaultPageTest.java]: Maybe we should use a string constant for site URL?

File: src/test/java/hw2/ex1/DefaultPageTest.java [hunk 12]
```diff
@@ -0,0 +1,124 @@
...
+    private WebDriver driver;
+
+    @BeforeMethod(alwaysRun = true)
+    public void ChromeSetUp()
+    {
+        driver = new ChromeDriver();
+    }
+
+    @Test (description = "Default Page Test")
+    public void MyTes_01() {
```
Comment (komyagin) [inline src/test/java/hw2/ex1/DefaultPageTest.java]: Very bad name for the method, not informative.

File: src/test/java/hw2/ex2/DifElementsPageTest.java [hunk 13]
```diff
@@ -0,0 +1,92 @@
...
+    private WebDriver driver;
+
+    @BeforeMethod(alwaysRun = true)
+    public void ChromeSetUp()
+    {
+        driver = new ChromeDriver();
+    }
+
+    @Test (description = "Different Elements Page Test")
+    public void MyTest_02()  {
```
Comment (komyagin) [inline src/test/java/hw2/ex2/DifElementsPageTest.java]: Very bad name for the method, not informative.

**Review by komyagin — CHANGES_REQUESTED** [review]

File: src/test/java/hw2/ex1/DefaultPageTest.java (hunk 11 above)
Comment (NikonovaEkaterina) [reply src/test/java/hw2/ex1/DefaultPageTest.java]: string constant added

File: src/test/java/hw2/ex2/DifElementsPageTest.java (hunk 13 above)
Comment (NikonovaEkaterina) [reply src/test/java/hw2/ex2/DifElementsPageTest.java]: changed to DifferentElementsPageTestMethod

File: src/test/java/hw2/ex1/DefaultPageTest.java (hunk 12 above)
Comment (NikonovaEkaterina) [reply src/test/java/hw2/ex1/DefaultPageTest.java]: changed to DefaultPageTestMethod

File: src/test/java/hw2/ex2/DifElementsPageTest.java (hunk 9 above)
Comment (NikonovaEkaterina) [reply src/test/java/hw2/ex2/DifElementsPageTest.java]: entities name updated

File: src/test/java/hw2/ex1/DefaultPageTest.java (hunk 1 above)
Comment (NikonovaEkaterina) [reply src/test/java/hw2/ex1/DefaultPageTest.java]: move to separate method

File: src/test/java/hw2/ex1/DefaultPageTest.java (hunk 2 above)
Comment (NikonovaEkaterina) [reply src/test/java/hw2/ex1/DefaultPageTest.java]: move to separate method

File: src/test/java/hw2/ex1/DefaultPageTest.java (hunk 3 above)
Comment (NikonovaEkaterina) [reply src/test/java/hw2/ex1/DefaultPageTest.java]: move to separate method

File: src/test/java/hw2/ex1/DefaultPageTest.java (hunk 4 above)
Comment (NikonovaEkaterina) [reply src/test/java/hw2/ex1/DefaultPageTest.java]: move to separate method

File: src/test/java/hw2/ex1/DefaultPageTest.java (hunk 5 above)
Comment (NikonovaEkaterina) [reply src/test/java/hw2/ex1/DefaultPageTest.java]: move to separate class

File: src/test/java/hw2/ex2/DifElementsPageTest.java (hunk 10 above)
Comment (NikonovaEkaterina) [reply src/test/java/hw2/ex2/DifElementsPageTest.java]: move to string constant

File: src/test/java/hw2/ex2/DifElementsPageTest.java (hunk 6 above)
Comment (NikonovaEkaterina) [reply src/test/java/hw2/ex2/DifElementsPageTest.java]: moved to separate class

File: src/test/java/hw2/ex2/DifElementsPageTest.java (hunk 7 above)
Comment (NikonovaEkaterina) [reply src/test/java/hw2/ex2/DifElementsPageTest.java]: move to separate method

File: src/test/java/hw2/ex2/DifElementsPageTest.java (hunk 8 above)
Comment (NikonovaEkaterina) [reply src/test/java/hw2/ex2/DifElementsPageTest.java]: moved to separate method

**Review by komyagin — APPROVED** [review]
HW2 is accepted

**Pull request merged**